index.js

```diff
@@ -101,14 +101,21 @@ function csurf (options) {
       return token
     }
 
+    var submitted = value(req)
+
     // generate & set secret
     if (!secret) {
+      if (!ignoreMethod[req.method] && submitted) {
+        return next(createError(403, 'csrf token submitted but there is no active session', {
```

Comment:
I would either make the message much less specific, or add additional logic and errors. For example, I feel like this message has the following two issues:

1. This module has the ability to be used with the double-submit cookie pattern. In this case, the secret is not stored in a session at all, yet this error will still be triggered, likely misleading users, especially those who are using double submit cookies and also have a session system in their code.
2. The term "active session" may have different meaning to different people. I get a lot of folks at the `express-session` module, for example, who think "active session" is only when they set a user or other property on the session, but that is of course not the case -- simply having the session cookie is what makes it active.

If the logic around this is just as-is, a suggestion could be to make the error be `'no establish csrf secret'` and maybe like `ENOECSRFTOKEN` or such. This is just a suggestion.